bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/EdCompTargetList.java

```diff
@@ -251,16 +250,34 @@ private boolean selectionIsBaseTarget() {
         return selectedTarget().exists(st -> t.exists(st::equals));
     }
 
+    /**
+     * Ensure that there is at most one selection for this node in TargetSelection.
+     */
+    static void checkTargetSelectionConsistency(final TargetEnvironment env,
+                                                final ISPObsComponent node) {
+        final Option<SPTarget> tOpt = TargetSelection.getTargetForNode(env, node);
+        final Option<SPCoordinates> cOpt = TargetSelection.getCoordinatesForNode(env, node);
+        final Option<IndexedGuideGroup> gOpt = TargetSelection.getIndexedGuideGroupForNode(env, node);
+        if ((tOpt.isDefined() && (cOpt.isDefined() || gOpt.isDefined()))
+                || (cOpt.isDefined() && gOpt.isDefined()))
+            throw new RuntimeException("Illegal node state: more than one selection recorded for node");
```

Comment:
Perhaps we are missing a way to define the selection such that the unpossible cases cannot be represented?  An ADT with a case for none, target, coordinate, or group.  Then if you have a selection you know it is just one of those four cases.  You could add a fold method to handle logic that applies to each individual case.

I don't want to prolong this task any more though, and this at least will give us a warning in those cases where we check.

Reply:
When I initially started working on this, I began with:
```
Option<ImEither<ImEither<Coodinates, SPTarget>, IndexedGuideGroup>>
```
but that just seemed ridiculous and was far too hard to work with, so I began to implement a class, `OneOfThree` that was isomorphic to a nested `ImEither`, and was going to make:
```
Option<OneOfThree<Coordinates, SPTarget, IndexedGuideGroup>>
```
to represent the four possible selection types. Again, that just struck me as being too much, even though it would have captured the constraints in the type better than what I ended up with. (I was inspired by how Rob used `fold` over `Target` subclasses in `Target`, as you mention here.)

In the end, this seemed easiest and I was afraid of mucking with the code, so I went with it, although it is a shortcoming in that safety is not ensured by type.

Reply:
(`OneOfThree<R,S,T>` would of course require `First<R,S,T>`, `Second<R,S,T>`, and `Third<R,S,T>`.)

Reply:
Let's just go with this as is.  It's much better than it was before regardless. 👍

Reply:
Excellent. I pushed my last PR, which makes the target table work as we discussed (the base is disabled if it is linked to another target, and when attempting to select the base, the actual target serving as the base is selected), and Travis just said yes, so I'm going to take the terrifying step and merge this.